radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java

```diff
@@ -96,13 +96,29 @@
 public class RadixApplicationAPI {
 	private static final Logger LOGGER = LoggerFactory.getLogger(RadixApplicationAPI.class);
 
-	public static class Result {
+	public class Result {
 		private final ConnectableObservable<SubmitAtomAction> updates;
 		private final Completable completable;
 
-		private Result(ConnectableObservable<SubmitAtomAction> updates, Completable completable) {
+		private Result(ConnectableObservable<SubmitAtomAction> updates) {
 			this.updates = updates;
-			this.completable = completable;
+			this.completable = updates
+				.ofType(SubmitAtomStatusAction.class)
+				.lastOrError()
+				.flatMapCompletable(status -> {
+					if (status.getStatusNotification().getAtomStatus() == AtomStatus.STORED) {
+						return Completable.complete();
+					} else {
+						JsonElement data = status.getStatusNotification().getData();
```

Comment:
I know this is not a change, but it was something I reacted to earlier reading the code, it feels a little bit low level to be working with JsonElement at this stage, in RadixApplicationAPI. In this scope the JsonElement is translated into an error, should we maybe try to create an enum holding errors and we working with those instead?

Reply:
yeah, good point, I'll add a TODO comment to fix this in the future